DependencyInjection/Compiler/TwigExceptionPass.php

```diff
@@ -26,5 +26,9 @@ public function process(ContainerBuilder $container)
         if ($container->has('fos_rest.exception_listener') && $container->has('twig.exception_listener')) {
             $container->removeDefinition('twig.exception_listener');
         }
+
+        if (!$container->has('templating.engine.twig')) {
+            $container->removeDefinition('fos_rest.exception.twig_controller');
```

Comment:
Indeed, the `ControllerResolver` class from the Symfony FrameworkBundle [lazy loads](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php#L66) controllers as services at runtime. For this, the service must not be `private` (similar to what applies to event listeners and form types). I opened #1296 as a hotfix to not keep the `master` branch in a broken state until we decide how to proceed with this pull request.